modules/phase_field/src/userobjects/DiscreteNucleationMap.C

```diff
@@ -73,26 +78,31 @@ DiscreteNucleationMap::execute()
       Real r, rmin = std::numeric_limits<Real>::max();
 
       // find the distance to the closest nucleus
+      Real local_radius = 0.0;
       for (unsigned i = 0; i < _nucleus_list.size(); ++i)
       {
         // use a non-periodic or periodic distance
         r = _periodic < 0
-                ? (_q_point[qp] - _nucleus_list[i].second).norm()
-                : _mesh.minPeriodicDistance(_periodic, _q_point[qp], _nucleus_list[i].second);
+                ? (_q_point[qp] - _nucleus_list[i].center).norm()
+                : _mesh.minPeriodicDistance(_periodic, _q_point[qp], _nucleus_list[i].center);
         if (r < rmin)
+        {
           rmin = r;
+          // grab the radius of the nucleus that this qp is closest to
```

Comment:
This will not work, because you can be closest to a small particle (and be on its interface) but far from a super large nucleus that overlaps the current point completely. So you will end up with interface discontinuities inside large nuclei.
Unfortunately you will have to compute the max of all intensity values for all nuclei overlapping the current point.